src/js/workshop-comments.js

```diff
@@ -0,0 +1,144 @@
+const commentForm = document.getElementById("comment-form");
+
+const nameDOMElement = document.getElementById("name");
+const commentDOMElement = document.getElementById("comment");
+
+commentForm.addEventListener("submit", (event) => {
+	event.preventDefault();
+
+	const nameDOMElement = document.getElementById("name");
+	const name = nameDOMElement.value.trim();
+	const commentDOMElement = document.getElementById("comment");
+	const comment = commentDOMElement.value.trim();
+
+	const submittedCommentMessage = document.querySelector(".post-failure-message");
+
+	disableFields(true);
+
+	// Hide required message
+	// Note: It's necessary to hide "required" indicators programmatically even though they have been hidden initially by CSS
+	// in the case the same user submits more comments after submitting an initial comment
+	document.querySelectorAll(".required").forEach(element => {
+		applyVisibility(element, false);
+	});
+	// Hide comment submission message
+	// Note: It's necessary to hide succesful/failed submitted comment message indicators programmatically even though they have been hidden initially by CSS
+	// in the case the same user submits more comments after submitting an initial comment
+	document.querySelectorAll(".save-response").forEach(element => {
+		applyVisibility(element, false);
+	});
+
+	// Verify input Fields
+	if ((name === "") & (comment === "")) {
+		document.querySelectorAll(".required").forEach(element => {
+			applyVisibility(element, true, "inline");
+		});
+		disableFields(false);
+		return;
+	}
+
+	if (name === "") {
+		applyVisibility(document.querySelector(".required-name"), true, "inline");
+		disableFields(false);
+		return;
+	}
+
+	if (comment === "") {
+		applyVisibility(document.querySelector(".required-comment"), true, "inline");
+		disableFields(false);
+		return;
+	}
+
+	const XHR = new XMLHttpRequest();
+
+	const commentForm = document.getElementById("comment-form");
+	// Bind the FormData object and the form element
+	const commentFormData = new FormData( commentForm );
+
+	// submit and add comment to top of comment section of page on successful data submission
+	XHR.addEventListener( "load", function() {
+		if (XHR.status === 200) {
+			setSubmittedComment();
```

Comment:
It's probably better to rename "setSubmittedComment" to "showSubmittedComment".

Reply:
Please move line 121 that shows the success message out of `setSubmittedComment()` function to right calling this function so the function doesn't perform more that what it's supposed to do and the logic here is cleaner.